generator_process/actions/convert_original_stable_diffusion_to_diffusers.py

```diff
@@ -0,0 +1,883 @@
+                text_model_dict[key[len("cond_stage_model.transformer.") :]] = checkpoint[key]
+
+        # load clip vision
+        model.model.load_state_dict(text_model_dict)
+
+        # load mapper
+        keys_mapper = {
+            k[len("cond_stage_model.mapper.res") :]: v
+            for k, v in checkpoint.items()
+            if k.startswith("cond_stage_model.mapper")
+        }
+
+        MAPPING = {
+            "attn.c_qkv": ["attn1.to_q", "attn1.to_k", "attn1.to_v"],
+            "attn.c_proj": ["attn1.to_out.0"],
+            "ln_1": ["norm1"],
+            "ln_2": ["norm3"],
+            "mlp.c_fc": ["ff.net.0.proj"],
+            "mlp.c_proj": ["ff.net.2"],
+        }
+
+        mapped_weights = {}
+        for key, value in keys_mapper.items():
+            prefix = key[: len("blocks.i")]
+            suffix = key.split(prefix)[-1].split(".")[-1]
+            name = key.split(prefix)[-1].split(suffix)[0][1:-1]
+            mapped_names = MAPPING[name]
+
+            num_splits = len(mapped_names)
+            for i, mapped_name in enumerate(mapped_names):
+                new_name = ".".join([prefix, mapped_name, suffix])
+                shape = value.shape[0] // num_splits
+                mapped_weights[new_name] = value[i * shape : (i + 1) * shape]
+
+        model.mapper.load_state_dict(mapped_weights)
+
+        # load final layer norm
+        model.final_layer_norm.load_state_dict(
+            {
+                "bias": checkpoint["cond_stage_model.final_ln.bias"],
+                "weight": checkpoint["cond_stage_model.final_ln.weight"],
+            }
+        )
+
+        # load final proj
+        model.proj_out.load_state_dict(
+            {
+                "bias": checkpoint["proj_out.bias"],
+                "weight": checkpoint["proj_out.weight"],
+            }
+        )
+
+        # load uncond vector
+        model.uncond_vector.data = torch.nn.Parameter(checkpoint["learnable_vector"])
+        return model
+
+
+    def convert_open_clip_checkpoint(checkpoint):
+        text_model = CLIPTextModel.from_pretrained("stabilityai/stable-diffusion-2", subfolder="text_encoder")
+
+        # SKIP for now - need openclip -> HF conversion script here
+        #    keys = list(checkpoint.keys())
+        #
+        #    text_model_dict = {}
+        #    for key in keys:
+        #        if key.startswith("cond_stage_model.model.transformer"):
+        #            text_model_dict[key[len("cond_stage_model.model.transformer.") :]] = checkpoint[key]
+        #
+        #    text_model.load_state_dict(text_model_dict)
+
+        return text_model
+    #endregion
+    
+    prediction_type = None
+    image_size = None
+    scheduler_type = "pndm"
+    extract_ema = False
+    pipeline_type = None
+    dump_path = os.path.join(DIFFUSERS_CACHE, os.path.splitext(os.path.basename(checkpoint_path))[0])
+
+    checkpoint = torch.load(checkpoint_path)
+    global_step = checkpoint["global_step"]
```

Comment:
A custom checkpoint I imported doesn't have this.
```suggestion
    global_step = checkpoint.get("global_step", None)
```